### PR TITLE
Fix crash on PAC evaluation error

### PR DESCRIPTION
**Pull request by abbradar**
Fixes https://github.com/qutebrowser/qutebrowser/issues/2476. Also add a test.

Haven't run the test because I have new Qt; let's see what Travis has to say.

### CODE REVIEW

**Comment by abbradar** [comment]
Hm, can I somehow mark a test that _doesn't_ need `fail_on_logging` applied? Or is this a bad way to solve this problem?

File: tests/unit/browser/webkit/network/test_pac.py [hunk 1]
```diff
@@ -250,5 +240,37 @@ def serve():
...
+                    reason="Segfaults when run with QtWebEngine tests on Linux")
+def test_fetch_evalerror():
+    test_str = """
+        function FindProxyForURL(domain, host) {
+            return "FOO";
+        }
+    """
+
+    res = fetcher_test(test_str)
+    proxies = res.resolve(QNetworkProxyQuery(QUrl("https://example.com/test")))
```
Comment (The-Compiler) [inline tests/unit/browser/webkit/network/test_pac.py]: You'll need to add a `caplog` argument to this test, and put this line in a  `with caplog.at_level(logging.ERROR):` block - then it won't fail the tests.

**Comment by abbradar** [comment]
Fixed; seems to work now (`py34-cov` test failure seems unrelated). Sorry for taking time, life got in the way.

**Comment by The-Compiler** [comment]
No worries! LGTM, will merge once the tests pass. I restarted the `py34-cov` env.

**Comment by The-Compiler** [comment]
Thank you!

**Pull request merged**